sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2790,6 +2780,9 @@ def _exec_code_on_head(
             mkdir_code = (f'{cd} && mkdir -p {remote_log_dir} && '
                           f'touch {remote_log_path}')
             code = job_lib.JobLibCodeGen.queue_job(job_id, job_submit_cmd)
+            if extra_cmd is not None:
+                code = f'{code} && {extra_cmd}'
```

Comment:
Seems not clean (e.g., the caller probably has no idea / can't remember why this arg is appended to job_lib.JobLibCodeGen.queue_job()'s result). Better way/names?

Reply:
Just refactored the code and added a comment. PTAL. Thanks!